### DISCUSSION TITLE
Global feature toggles

### DISCUSSION THREAD

**Discussion by markosca** [Unleash/unleash#4333] [Ideas]
Hello team,

firstly I'd to thank you for your work. I'm currently using Unleash and the new features you have implemented are really nice.

However, I work in a complex environment where some projects have cross-dependencies. Then, sometimes we cannot enable a feature toggle until another one has been properly activated. 
Considering this "problem, " I have tried to configure a feature toggle setting other as a dependency but haven't been able... Is there any way to do it or should it be a new feature?

Thanks!
Marcos.

**Comment by thomasheartman** [comment]
Hey, @markosca 😄 Thanks for the compliments and sorry for the slow response 🙇🏼 

As for your question regarding configuring flag dependencies: we don't directly support using other flags as dependencies just yet, but we're working on it right now (tracked in [this issue](https://github.com/Unleash/unleash/issues/2255 "dependent feature flags"))! 

However, the first iteration will not support cross-project dependencies, so it doesn't actually address the issue you're describing here, but we'll take note of that as a feature request 📝 

That said, there might be a workaround. You can't specify the dependency in the flag configuration directly, but you can do it in code. For instance, in JavaScript, it might look something like this:

```js
if (unleash.isEnabled('parent') && unleash.isEnabled('child') {
  // do something
}
```

Would that solve your issue for the time being?

**Reply by markosca** [reply]
Thanks for your reply @thomasheartman!

I was thinking about the workaround you propose but obviously is not the best solution... it's just a workaround.

We have many components related and will be really important to have those global Feature flags but I understand you have many topics to work on. In any case, thanks for taking not and I hope you will see it as important as I.

Thanks for your support!
Marcos

**Reply by thomasheartman** [reply]
Of course! And yes, we've registered it in our feature requests database, so we'll prioritize it when we get there. 

There was also another solution that was suggested. When dependent flags are available, you could potentially get around the cross-project dependencies by moving all flags that have cross-project dependencies into the same project. That should let Unleash take care of dependencies directly, but it does come at a slight organizational cost. Depending on how your instance is set up and how your company uses Unleash, it may or may not be worth it.

As always, any more questions, just let me know 😄

**Reply by ivarconr** [reply]
This feature is now GA: https://docs.getunleash.io/reference/dependent-features
